src/storage/src/source/mysql/replication.rs

```diff
@@ -178,10 +178,29 @@ pub(crate) fn render<G: Scope<Timestamp = GtidPartition>>(
 
             trace!(%id, "timely-{worker_id} replication binlog frontier: {binlog_frontier:?}");
 
-            // Calculate the minimum frontier across all subsources, which represents the point which
+            // Calculate the lowest frontier across all subsources, which represents the point which
             // we should start replication from.
             let resume_upper =
-                Antichain::from_iter(subsource_resume_uppers.into_values().flatten());
+                match Antichain::from_iter(subsource_resume_uppers.into_values().flatten()) {
+                    upper if upper == Antichain::from_elem(GtidPartition::minimum()) => {
+                        // If any subsource is at the minimum frontier then we are either starting
+                        // from scratch or at least one table has to complete its initial snapshot.
+                        //
+                        // In either case, we need to ensure that the resumption point is
+                        // deterministic and consistent across all tables, so that a user can't
+                        // observe an MZ timestamp that may contain different snapshots for
+                        // different tables.
+                        // We've chosen the frontier beyond the GTID Set recorded
+                        // during purification as this consistent resume point.
+                        //
+                        // Tables may still be snapshot at a future GTID Set but as long as we can
+                        // resume the replication stream from this consistent point all updates
+                        // will be rewound appropriately, such that the effective snapshot point
+                        // is consistent across all tables.
+                        gtid_set_frontier(connection.details.initial_gtid_set.as_ref())?
```

Comment:
rust tip: if you have a `String` and a function that accepts `&str` you can pass `&foo` and rust will auto-dereference the value as many times as possible until it finds the appropriate type. so in this case this could be `&connection.details.initial_gtid_set`